packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -63,7 +63,7 @@ function DialogTrigger(props: SpectrumDialogTriggerProps) {
         console.warn('A DialogTrigger unmounted while open. This is likely due to being placed within a trigger that unmounts or inside a conditional. Consider using a DialogContainer instead.');
       }
     };
-  }, []);
+  }, [type]);
```

Comment:
Is this in case it switches between types? is there a test case or storybook I should be looking at?

Reply:
This is once again for an ESLint warning.